flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/FlinkFilters.java

```diff
@@ -246,18 +248,70 @@ private static Optional<Expression> convertFieldAndLiteral(
     org.apache.flink.table.expressions.Expression left = args.get(0);
     org.apache.flink.table.expressions.Expression right = args.get(1);
 
-    if (left instanceof FieldReferenceExpression && right instanceof ValueLiteralExpression) {
-      String name = ((FieldReferenceExpression) left).getName();
-      Optional<Object> lit = convertLiteral((ValueLiteralExpression) right);
+    Optional<Object> lit;
+    if (left instanceof FieldReferenceExpression) {
+      lit = convertExpression(right);
       if (lit.isPresent()) {
-        return Optional.of(convertLR.apply(name, lit.get()));
+        return Optional.of(convertLR.apply(((FieldReferenceExpression) left).getName(), lit.get()));
```

Comment:
I don't see a reason to remove the `name` variable in this block or the mirror one. That introduces needless changes. Can you roll that change back?